src/gui/osutils/winutils/WinUtils.cpp

```diff
@@ -79,6 +96,41 @@ bool WinUtils::nativeEventFilter(const QByteArray& eventType, void* message, lon
     case WM_HOTKEY:
         triggerGlobalShortcut(msg->wParam);
         break;
+    case WM_NCCREATE:
+    case WM_ENTERIDLE: // modal dialog box i.e.: save file dialog
+        if (not allowScreenCapture()) {
+            if (msg->message != WM_ENTERIDLE || DWORD(msg->wParam) == MSGF_DIALOGBOX) {
+                HWND hwnd = msg->message == WM_ENTERIDLE ? (HWND)msg->lParam : msg->hwnd;
+                if (not SetWindowDisplayAffinity(hwnd, WDA_EXCLUDEFROMCAPTURE)) {
+                    qCritical().nospace() << "Couldn't set WDA_EXCLUDEFROMCAPTURE display affinity flag to window. error_code: 0x"
+                                          << hex << GetLastError();
```

Comment:
This is the only relevant change, and the only desired change. Remove everything else.

Reply:
I cleaned the code, please verify if anything else should be changed.